src/deltacode/utils.py

```diff
@@ -179,40 +189,36 @@ def deltas(deltacode, all_delta_types=False):
     """
     for delta in deltacode.deltas:
         if all_delta_types is True:
-            yield delta.to_dict()
+            yield delta.to_dict(deltacode)
         elif not delta.is_unmodified():
-            yield delta.to_dict()
+            yield delta.to_dict(deltacode)
 
 def calculate_percent(value, total):
     """
     Return the rounded value percentage of total.
     """
-    ratio = (value / total) * 100
-    return round(ratio, 2)
+    try :
+        ratio = (value / total) * 100
+        return round(ratio, 2)
+    except ZeroDivisionError:
+        return 0
 
 class AlignmentException(Exception):
     """
     Named exception for alignment errors.
     """
     pass
 
-def align_trees(a_files, b_files):
-    """
-    Given two sequences of File objects 'a' and 'b', return a tuple of
-    two integers that represent the number path segments to remove
-    respectively from a File path in 'a' or a File path in 'b' to obtain the
-    equal paths for two files that are the same in 'a' and 'b'.
-    """
-    # we need to find one uniquly named file that exists in 'a' and 'b'.
+def align_trees(codebase1, codebase2):
```

Comment:
Do we still need this function?

Reply:
I thaught of removing align trees , but then our scoring system is based upon after aligning the path , so I planned to keep it.

Reply:
Can you point me to the line (s) of code that uses path for scoring?

Reply:
https://github.com/nexB/deltacode/blob/bf767c2a05ca12f40d77036a6f30375f9fbf1c0e/src/deltacode/__init__.py#L107

We are basically creating deltas after aligning of paths(based on the aligned path ) in old architecture and 
after the creation of the delta we assign scores based on `removed/added...` and all the test cases/expected results are in the Design are based upon it too.
So i thaught of keeping the alignment for now

https://github.com/nexB/deltacode/blob/bf767c2a05ca12f40d77036a6f30375f9fbf1c0e/src/deltacode/__init__.py#L122

The `path` which we are enumerating actually refers to the aligned path

Reply:
But does this work with virtual codebase?

Reply:
Not directly actually as we are having the `original_path`  and to alling the path the path we just need two `OFFSET` for path aligning